src/data/i18n.tsx

```diff
@@ -3,7 +3,7 @@ import { findDefaultingToHead } from '../utils/helpers'
 export const LOCALES = ['en', 'de'] as const
 export type Locale = typeof LOCALES[number]
 
-type WithLocale<T> = T & { locale: string }
+type WithLocale<T> = { value: T; locale: string }
```

Comment:
This is the important part. Everything else is a consequence of this

Reply:
I like that simple change and what features it brings with it :smiley: